FileIO/GmshIO/GMSHAdaptiveMeshDensity.cpp

```diff
@@ -141,13 +143,13 @@ void GMSHAdaptiveMeshDensity::getQuadTreeGeometry(std::vector<GeoLib::Point*> &p
 	for (std::list<GeoLib::QuadTree<GeoLib::Point>*>::const_iterator it(leaf_list.begin()); it
 		!= leaf_list.end(); ++it) {
 		// fetch corner points from leaf
-		GeoLib::Point *ll(new GeoLib::Point), *ur(new GeoLib::Point);
-		(*it)->getSquarePoints(*ll, *ur);
-		std::size_t pnt_offset (pnts.size());
-		pnts.push_back(ll);
-		pnts.push_back(new GeoLib::Point((*ur)[0], (*ll)[1], 0.0));
-		pnts.push_back(ur);
-		pnts.push_back(new GeoLib::Point((*ll)[0], (*ur)[1], 0.0));
+		GeoLib::Point ll, ur;
+		(*it)->getSquarePoints(ll, ur);
+		std::size_t const pnt_offset (pnts.size());
+		pnts.push_back(new GeoLib::Point(ll, pnt_offset));
```

Comment:
not sure but if you don't correct the idx in my other comment you might end up with two points having idx ```pnt_offset``` (i might be wrong, though, i don't know how these two methods interact / in which sequence they are called.